public/app/features/dashboard-scene/panel-edit/PanelDataPane/PanelDataQueriesTab.tsx

```diff
@@ -358,9 +352,35 @@ export function PanelDataQueriesTabRendered({ model }: SceneComponentProps<Panel
             {queryLibraryEnabled && (
               <Button
                 icon="plus"
-                onClick={() => {
-                  openQueryLibraryDrawer(getDatasourceNames(datasource, queries), addQueryActionButton);
-                }}
+                onClick={() =>
+                  openQueryLibraryDrawer(getDatasourceNames(datasource, queries), async (query) => {
```

Comment:
Nice, works well. Sort of nitpicky but I would probably move it to separate method cause it can look confusing that this code is 2 separate `onClick` handlers.